docs/DevelopmentGuide.md

```diff
@@ -69,14 +92,33 @@ bundle exec rake test:main
 
 #### For integrations
 
-Integrations which interact with dependencies not listed in the `datadog-ci` gemspec will need to load these dependencies to run their tests.
+Integrations which interact with dependencies not listed in the `datadog-ci` gemspec will need to load these dependencies to run their tests. Each test task could consist of multiple spec tasks which are executed with different groups of dependencies (likely against different versions or variations).
 
 To get a list of the spec tasks run `bundle exec rake -T 'test:'`
 
 To run any of the specs above run `bundle exec rake 'test:<spec_name>'`.
 
 For example: `bundle exec rake test:minitest`
 
+#### Working with appraisal groups
+
+Checkout [Apppraisal](https://github.com/thoughtbot/appraisal) to learn the basics.
+
+Groups are defined in the `Apparisals` file and their names are prefixed with Ruby runtime based on the environment. `*.gemfile` and `*.gemfile.lock` from `gemfiles/` directory are generated from those definitions.
+
+To find out existing groups in your environment, run `bundle exec appraisal list`
+
+After introducing a new group definition or changing existing one, run `bundle exec appraisal generate` to propagate the changes.
+
+To install dependencies, run `bundle exec appraisal install`.
+
+In addition, if you already know which appraisal group definition to work with, you can target a specific group operation with environment vairable `APPRAISAL_GROUP`, instead of all the groups from your environment. For example:
+
+```bash
+# This would only install dependencies for `rspec` group definition
+APPRAISAL_GROUP=rspec bundle exec appraisal install
```

Comment:
I think `rspec` is not a good example, because it is declared from the `Gemfile` at the project root. 
Perhaps using `cucumber` make more sense

Reply:
true, will replace with cucumber